src/kubernetes.cc

```diff
@@ -961,6 +961,33 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidateConfiguration() const {
+  try {
+    (void) QueryMaster(std::string(kKubernetesEndpointPath) + "/nodes?limit=1");
+  } catch (const QueryException& e) {
+    // Already logged.
+    return false;
+  }
+
+  try {
+    const std::string pod_label_selector(
+        config_.KubernetesPodLabelSelector().empty()
+        ? "" : "&" + config_.KubernetesPodLabelSelector());
+
+    (void) QueryMaster(std::string(kKubernetesEndpointPath) + "/pods?limit=1" +
+                       pod_label_selector);
```

Comment:
Same question as for Docker -- can we confirm that the first `limit=1` will take precedence over whatever is in the selector?

Reply:
`pod_label_selector` does not imply passing a limit, it implies passing labels, if a user is passing a limit there, that's a hack, and should be validated elsewhere.

Reply:
I agree that it's a hack, but a `limit=500000` in the (user-specified) selector should not cause us to suddenly grab all the pods in the cluster. Hence I was just asking whether this `limit=1` will override it where it is now, or whether it should be at the end.

That said, I've just verified that the first occurrence of `limit` takes precedence, so we're good.